components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt

```diff
@@ -196,17 +197,17 @@ class GeckoEngine(
         supportActions: Boolean,
         onSuccess: ((WebExtension) -> Unit),
         onError: ((String, Throwable) -> Unit)
-    ) {
+    ): CancellableOperation {
         val ext = GeckoWebExtension(id, url, runtime, allowContentMessaging, supportActions)
-        installWebExtension(ext, onSuccess, onError)
+        return installWebExtension(ext, onSuccess, onError)
     }
 
     internal fun installWebExtension(
         ext: GeckoWebExtension,
         onSuccess: ((WebExtension) -> Unit) = { },
         onError: ((String, Throwable) -> Unit) = { _, _ -> }
-    ) {
-        if (ext.isBuiltIn()) {
+    ): CancellableOperation {
+        val geckoResult = if (ext.isBuiltIn()) {
```

Comment:
nit: `return if`?  Although arguably this is more readable :).

Reply:
Yeah, let's leave it, it gets harder to read otherwise ;).